library/src/main/java/com/novoda/downloadmanager/lib/DownloadNotifier.java

```diff
@@ -327,20 +332,38 @@ private Notification buildTitlesAndDescription(
 
     }
 
-    private Notification buildSingleNotification(int type, NotificationCompat.Builder builder, BatchInfo batch, String remainingText, String percentText) {
+    /**
+     * Return given duration in a human-friendly format. For example, "4
+     * minutes" or "1 second". Returns only largest meaningful unit of time,
+     * from seconds up to hours.
+     */
+    public CharSequence formatDuration(long millis) {
+        if (millis >= DateUtils.HOUR_IN_MILLIS) {
+            int hours = (int) TimeUnit.MILLISECONDS.toHours(millis + TimeUnit.MINUTES.toMillis(30));
+            return resources.getQuantityString(R.plurals.dl__duration_hours, hours, hours);
+        } else if (millis >= DateUtils.MINUTE_IN_MILLIS) {
+            int minutes = (int) TimeUnit.MILLISECONDS.toMinutes(millis + TimeUnit.SECONDS.toMillis(30));
+            return resources.getQuantityString(R.plurals.dl__duration_minutes, minutes, minutes);
+        } else {
+            int seconds = (int) TimeUnit.MILLISECONDS.toSeconds(millis + 500);
+            return resources.getQuantityString(R.plurals.dl__duration_seconds, seconds, seconds);
+        }
+    }
+
+    private Notification buildSingleNotification(int type, NotificationCompat.Builder builder, DownloadBatch batch, String remainingText, String percentText) {
 
         NotificationCompat.BigPictureStyle style = new NotificationCompat.BigPictureStyle();
-        String imageUrl = batch.getBigPictureUrl();
+        String imageUrl = batch.getInfo().getBigPictureUrl();
```

Comment:
might look into `batch.getBigPictureUrl()`  "Only talk to your closest neighbour." , not for this PR